frontend/client/views/tools/Wrapper.vue

```diff
@@ -23,9 +172,181 @@ export default {
       message: 'This page doesn\'t work.',
       type: 'warning'
     })
+
+    // fetch csrf token upon mounting
+    this.$http.get('/api/wrapping')
+    .then((response) => {
+      this.csrf = response.headers['x-csrf-token']
+    })
+    .catch((error) => {
+      this.$onError(error)
+    })
   },
 
   methods: {
+
+    // takes out "isClicked" field in tableData so content can be sent off
+    packData: function () {
+      var data = {}
+      for (var i = 0; i < this.tableData.length; i++) {
+        data[this.tableData[i].key] = this.tableData[i].value
+      }
+      return data
+    },
+
+    wrapData: function () {
+      // do nothing if the table is empty
+      if (this.tableData.length === 0) {
+        return
+      }
+
+      this.$http.post('/api/wrapping/wrap', querystring.stringify({
+
+        // wrapttl takes value of user's input, default at 300s
+        wrapttl: this.wrap_ttl,
+        data: JSON.stringify(this.packData())
+      }), {
+        headers: {'X-CSRF-Token': this.csrf}
+      })
+      .then((response) => {
+        // wrapping token:
+        this.$message({
+          message: 'Your token is: ' + response.data.result,
+          type: 'success',
+          duration: 0,
+          showCloseButton: true
+        })
+        console.log(response.data.result)
+      })
+      .catch((error) => {
+        this.$onError(error)
+      })
+    },
+
+    unWrapToken: function () {
+      // do nothing if there is no input token string
+      if (this.currToken === '') {
+        return
+      }
+      this.$http.post('/api/wrapping/unwrap', querystring.stringify({
+        wrappingToken: this.currToken
+      }), {
+        headers: {'X-CSRF-Token': this.csrf}
+      })
+      .then((response) => {
+      // unwrapped data:
+        this.tableData = []
+        this.unpackData(response.data.result)
+        console.log(response.data.result)
+      })
+      .catch((error) => {
+        this.$onError(error)
+      })
+    },
+
+    unpackData: function (rawTable) {
+      Object.keys(rawTable).map((index) => this.tableData.push({
+        key: index,
+        value: rawTable[index],
+        isClicked: false
+      }))
+    },
+
+    deleteItem: function (index) {
+      this.tableData.splice(index, 1)
+    },
+
+    clicked: function (index) {
+      this.tableData[index].isClicked = true
+    },
+
+    // Returns true if the new key already exists in the current table
+    newKeyExists: function () {
```

Comment:
This function should take a key value as an argument, and search through tableData for key conflicts. Should also rename it to `keyExists` in that case.